src/java/org/apache/cassandra/auth/CassandraAuthorizer.java

```diff
@@ -131,6 +151,8 @@ public void revokeAllFrom(RoleResource revokee)
                                                        ClientState.forInternalCalls()));
 
             executeLoggedBatch(statements);
+            
+            invalidateRolesAndPermissions(Collections.singleton(revokee));
```

Comment:
This `revokeAll` is called when a role is dropped. 

Here, the `invalidateRolesAndPermissions` being called delegates to `AuthenticatedUser.invalidate(Collection<RoleResource> roles)` to remove all permission cache entries associated with the role (using a filter on part of the key). This is also called from `CassandraRoleManager` during grant or revoke role operations.

This `revokeAll` method has a query to find all the resources associates with a role; so an alternative approach would be to collect those and then iteratively call `invalidateRolesAndPermissions(Collection<RoleResource> roles, IResource resource)`.

I decided not to do it that way because, a) the method in `AuthenticatedUser` to invalidate using a filter by role is needed anyway by `CassandraRoleManager` and b) it seemed better to invalidate whatever is currently in the cache in case there would be any discrepancy in resources that the query in `revokeAll` returns; or why rely on it when we can simply invalidate all role entries in the cache already.